media/com_templates/js/admin-template-compare.es6.js

```diff
@@ -0,0 +1,143 @@
+        pre.appendChild(document.createTextNode(decodeHtmlspecialChars(part.value)));
+        fragment.appendChild(pre);
+      });
+
+      display.appendChild(fragment);
+    };
+
+    const buttonDataSelector = 'onclick-task';
+    const override = document.getElementById('override-pane');
+    const corePane = document.getElementById('core-pane');
+    const diffMain = document.getElementById('diff-main');
+
+    const toggle = (e) => {
+      const task = e.target.getAttribute(buttonDataSelector);
+      if (task === 'template.show.core') {
+        if (corePane) {
+          const { display } = corePane.style;
+          if (display === 'none') {
+            e.target.className = 'btn btn-success';
+            e.target.innerHTML = Joomla.JText._('COM_TEMPLATES_LAYOUTS_DIFFVIEW_HIDE_CORE');
+            corePane.style.display = 'block';
+            override.className = 'col-md-6';
+          } else {
+            e.target.className = 'btn btn-danger';
+            e.target.innerHTML = Joomla.JText._('COM_TEMPLATES_LAYOUTS_DIFFVIEW_SHOW_CORE');
+            corePane.style.display = 'none';
+            override.className = 'col-md-12';
+          }
+          const coreState = {
+            class: e.target.className,
+            title: e.target.innerHTML,
+            display: corePane.style.display,
+            overrideClass: override.className,
+          };
+
+          if (typeof (Storage) !== 'undefined') {
+            localStorage.setItem('coreButtonState', JSON.stringify(coreState));
+          }
+        }
+      } else if (task === 'template.show.diff') {
+        if (diffMain) {
+          const { display } = diffMain.style;
+          if (display === 'none') {
+            e.target.className = 'btn btn-success';
+            e.target.innerHTML = Joomla.JText._('COM_TEMPLATES_LAYOUTS_DIFFVIEW_HIDE_DIFF');
+            diffMain.style.display = 'block';
+          } else {
+            e.target.className = 'btn btn-danger';
+            e.target.innerHTML = Joomla.JText._('COM_TEMPLATES_LAYOUTS_DIFFVIEW_SHOW_DIFF');
+            diffMain.style.display = 'none';
+          }
+          const diffState = {
+            class: e.target.className,
+            title: e.target.innerHTML,
+            display: diffMain.style.display,
+          };
+
+          if (typeof (Storage) !== 'undefined') {
+            localStorage.setItem('diffButtonState', JSON.stringify(diffState));
+          }
+        }
+      }
+    };
+
+    const buttons = [].slice.call(document.querySelectorAll(`[${buttonDataSelector}]`));
+    const conditionalSection = document.getElementById('conditional-section');
+
+    const setPrestate = () => {
+      if (typeof (Storage) !== 'undefined') {
+        const cState = JSON.parse(localStorage.getItem('coreButtonState'));
+        const dState = JSON.parse(localStorage.getItem('diffButtonState'));
+        buttons[0].className = cState.class;
+        buttons[0].innerHTML = cState.title;
```

Comment:
I think without class and title it looks bad. Because, we are showing different state with color and title . like
![screenshot from 2018-06-06 15-37-33](https://user-images.githubusercontent.com/30978328/41031921-93f2c1e2-699f-11e8-8aac-258380ea47ea.png)
![screenshot from 2018-06-06 15-37-14](https://user-images.githubusercontent.com/30978328/41031922-94d493ce-699f-11e8-999a-116fb7932c4d.png)

Reply:
Ok, we can try with a switcher later. Because currently, all things are working fine. And we can fix UI later. What do you think ?   @astridx @laoneo

Reply:
If this got merge then I can proceed to next phase.

Reply:
@laoneo Are you Ok if create an issue for changing the buttons to a switch. If we have time at the end or meanwhile - when we have to make some resource - we can do it.

Reply:
Yes, issue is fine, Just want to make sure we are not forgetting it.

Reply:
https://github.com/joomla-projects/gsoc18_override_management/issues/13